### DISCUSSION TITLE
Is Mapbox SDK v10 a possibility for this library?

### DISCUSSION THREAD

**Discussion by nzjs** [rnmapbox/maps#1202] [Q&A] [answered]
Mapbox SDK v10 has just been released, which includes the new 3D terrain and greatly improved map performance (apparently) along with some other features. 
[https://www.mapbox.com/blog/mapbox-mobile-sdk-v10-beta](https://www.mapbox.com/blog/mapbox-mobile-sdk-v10-beta) 
[https://www.mapbox.com/mapbox-mobile-sdk](https://www.mapbox.com/mapbox-mobile-sdk)

Is this something that this library can work towards implementing? 
If so, what can we do to help out?

I would love to be able to offer 3D terrain in my apps.

**Comment by systemlevel** [comment]
@ferdicus @mfazekas @kristfal Would love to hear your thoughts, in particular, on the feasibility of being able to support this.

**Reply by ferdicus** [reply]
hey @systemlevel , I mean I guess we have to eventually move in that direction as I would imagine, that mapbox is gonna phase out the older SDKs. 
Having said that, I have one day per week for this repo - I can barely merge the dependency updates 😅 

It would be interesting to know if there are also licensing changes with the update, as I know, that there were some rumblings concerning this during the 5 => 6 transition.

Plus (there is even more!) - these are written in Kotlin and Swift?! 😓 😅 😓

**Reply by systemlevel** [reply]
@ferdicus Cool, yep I totally get the time availability issue. My question was less about "hey can you work on this" and more like based on your technical understanding is this a feasibility somehow?  

Seems like based on your assessment this could entail a lot of rework, not to mention potentially having to use a new language like Swift?

**Reply by ferdicus** [reply]
yeah, think so - even if the actual work of editing the wrapper and such is not that involved. 
It would be nice to finally actually have some tests - but maybe that's just me 🤷🏿‍♂️ 

Again, this is from me literally not understanding a thing about native development and having looked at the newly set up repos for 5mins 😅

**Reply by systemlevel** [reply]
@ferdicus Yes. Thanks for your thoughts and thank you for your continued support of this repository.

**Reply by nzjs** [reply]
Thanks for the response. If someone eventually has time to plan out what is required for v10 along with a list of issues needing resolved, I am sure the community (including myself) would be willing to help with implementation.

**Reply by mfazekas** [reply]
I've played a bit with iOS version of the v10 sdk. There are significant api changes and we need to move to swift for sure. See [iOS migration guide](https://docs.mapbox.com/ios/beta/maps/guides/migrate-to-v10/).

I've just looked the android sdk, and while there are lot of changes, it sounds a simpler change, based on the [migration guide.](https://docs.mapbox.com/android/beta/maps/guides/migrate-to-v10/)

**Reply by systemlevel** [reply]
@mfazekas Would we need to do a total rewrite in swift or can we bridge the current Obj-C API with the new Swift features?  Not sure if that makes sense in the long run. Perhaps as a step in the process?

**Reply by mfazekas** [reply]
> @mfazekas Would we need to do a total rewrite in swift or can we bridge the current Obj-C API with the new Swift features? Not sure if that makes sense in the long run. Perhaps as a step in the process?

@systemlevel we can have objc code if we need to. But given much of our code is boilerplate, and just wraps the the Mapbox API, I don't think we'll use that, but we'll see.

**Reply by ferdicus** [reply]
To be honest, I'd vote for doing a hard cut (if feasible) instead of:
mapbox-SDK `Swift` => our native modules `Obj-C` => `TS` => `JS` 
have
mapbox-SDK `Swift` => our native modules `Swift` => `TS` => `JS` 

We can ditch one language 🎉  😞  🎉

**Reply by systemlevel** [reply]
Yes it seems like they are committing fully to swift moving forward. Maybe we pull off the bandaid at the get go and eliminate obj-c.

**Comment by systemlevel** [comment]
I know a company that is willing to contribute possible resources (financially mostly) to this effort.  Would there be any other companies we can reach out to to try and get something like this accomplished?

**Comment by mfazekas** [comment]
I've added a POC for the iOS part. This demonstrates the way it can be implemented using a mix of ObjC and Swift. But it requires significant rewrite.

https://github.com/react-native-mapbox-gl/maps/pull/1222#issue-574317825

**Comment by maxammann** [comment]
I just stared a similar discussion here: https://github.com/react-native-mapbox-gl/maps/discussions/1274

The new discussion is kind of against upgrading to v10.

**Comment by systemlevel** [comment]
I have a company that is interested in helping move this forward but would like to work with someone or another company in our community here. They have funding, etc. Please reach out if you are interested in either receiving payment for helping with this effort or have resources to put into it.

**Comment by maxammann** [comment]
Yey! This library supports MapLibre now! https://github.com/react-native-mapbox-gl/maps/blob/master/android/install.md#using-maplibre

**Reply by systemlevel** [reply]
Does anyone know or have insight into whether or not MapLibre is going to support 3D etc?

**Reply by maxammann** [reply]
I think it wont. At least im the near future.
Someone has to invest a lot of time to implement this.

**Reply by ferdicus** [reply]
~@systemlevel, what exactly do you mean by `3D`?~

Nevermind, you're talking about the 3D terrain feature that OP mentioned when talking about new v10 features 👍🏿

**Reply by systemlevel** [reply]
@ferdicus Yes, that's correct.  3D terrain V10.

**Comment by danidaryaweesh** [comment]
Any updates on this topic? are there any plans to support Mapbox SDK 10?

**Reply by nzjs** [reply]
There has been further discussion below. The main constraints seem to be a lack of funding and as such, availability from the devs that have knowledge of the work required.

https://github.com/react-native-mapbox-gl/maps/pull/1222#issue-574317825

**Comment by systemlevel** [comment, marked as answer]
Please see the following comments: https://github.com/react-native-mapbox-gl/maps/pull/1222#issuecomment-996221674

**Comment by ferdicus** [comment]
please see https://github.com/react-native-mapbox-gl/maps/discussions/1680 for a follow-up discussion
